apis/api_test.go

```diff
@@ -0,0 +1,107 @@
+package apis_test
+
+import (
+	"cloudfoundry.org/cf-k8s-api/apis"
+	. "github.com/onsi/gomega"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+const (
+	jsonHeader = "application/json"
+	defaultServerURL = "https://api.example.org"
+)
+func TestAPI(t *testing.T) {
+	spec.Run(t, "object", testRootV3API, spec.Report(report.Terminal{}))
+	spec.Run(t, "object", testRootAPI, spec.Report(report.Terminal{}))
+}
+
+func testRootV3API(t *testing.T, when spec.G, it spec.S) {
+	Expect := NewWithT(t).Expect
+
+	when("the v3 GET endpoint returns successfully", func() {
+		// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+		// pass 'nil' as the third parameter.
+		req, err := http.NewRequest("GET", "/v3", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// We create a ResponseRecorder (which satisfies http.ResponseWriter) to record the response.
+		rr := httptest.NewRecorder()
+		apiHandler := apis.RootV3Handler{
+			ServerURL: defaultServerURL,
+		}
+
+		handler := http.HandlerFunc(apiHandler.RootV3GetHandler)
+
+		// Our handlers satisfy http.Handler, so we can call their ServeHTTP method
+		// directly and pass in our Request and ResponseRecorder.
+		handler.ServeHTTP(rr, req)
+
+		it("returns status 200 OK", func() {
+			httpStatus := rr.Code
+			Expect(httpStatus).Should(Equal(http.StatusOK), "Matching HTTP response code:")
+		})
+		it("returns Content-Type as JSON in header", func() {
+			contentTypeHeader := rr.Header().Get("Content-Type")
+			Expect(contentTypeHeader).Should(Equal(jsonHeader), "Matching Content-Type header:")
+		})
+		it("has a non-empty body", func() {
```

Comment:
Minor stylistic note (feel free to disagree):

I recognize the value this test added during the TDD process, but I think assertions like this one can be rolled up into the other assertion on response body content. Like the following assertion:

```
Expect(rr.Body.String()).Should(Equal(expectedBody), "Response body matches RootV3GetHandler response:")
```

This implicitly tests that the Body is not empty. This can cut down on test suite bloat.

Reply:
Valid point. I have addressed this